packtools/sps/models/article_assets.py

```diff
@@ -51,3 +51,16 @@ def __init__(self, node, parent_map):
     def name(self):
         return self.node.attrib["{http://www.w3.org/1999/xlink}href"]
 
+    @property
+    def id(self):
+        current_node = self.node
+
+        while current_node is not None and hasattr(current_node, 'attrib') and 'id' not in current_node.attrib:
+            current_node = self.parent_map.get(current_node)
+
+        if not current_node or not hasattr(current_node, 'attrib'):
+            return
+
+        current_node_attrib = getattr(current_node, 'attrib')
+        if current_node_attrib:
+            return current_node_attrib.get('id')
```

Comment:
Me parece custoso.

Já testou com alternatives? Acho que vai falhar

Reply:
Não falha, pelo menos nos casos que eu availie e envie no PR.

Reply:
Para cada nó que é asset, o custo de descobrir o ID é equivalente ao tamanho de sua altura (distância até a raiz).

O método é recursivo e caminha de filho para pai, de pai para avô, de avô para ...; isso ocorre até encontrar um ID ou ao chegar na raiz. Essa é a motivação de termos um dict que mapeia Nó a Pai (para baratear a busca).

Isso faz com que saibamos o ID de um graphic que está dentro de um fig e o fig é quem tem o ID.
Na pior das hipóteses, ele chegará à raiz sem encontrar ID.

Isso só ocorre para os nós que são assets.

Reply:
Acho que IDs no meio do caminho de uma busca têm o potencial de atrapalhar o resultado.

Mas que outra forma haveria de encontrar o melhor ID de um graphic ou media...

O método começaria a ficar cheio de especialidades (só considerar ID se for um fig, etc...)